### PR TITLE
Add OwrBus which aggregates messages from a new OwrMessageOrigin interface

### PR DESCRIPTION
**Pull request by Rugvip**
WIP, but I'm keeping all commits buildable, so parts of it could be landed.

### CODE REVIEW

**Comment by Rugvip** [comment]
Updated with fix from previous comment about g_strdup everywhere

File: owr/owr_bus.h [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+typedef enum {
+    OWR_MESSAGE_TYPE_ERROR = (1 << 0),
+    OWR_MESSAGE_TYPE_STATS = (1 << 1),
+    OWR_MESSAGE_TYPE_EVENT = (1 << 2)
+} OwrMessageType;
+
+/**
+ * OwrMessageSubType:
+ *
+ * @OWR_ERROR_TYPE_GST_ERROR: a GStreamer error message was received
```
Comment (superdump) [inline owr/owr_bus.h]: Media processing error or something? It's not really interesting to the application that it's specifically from GStreamer. I guess.

File: owr/owr_bus.h [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+ */
+typedef enum {
+    OWR_ERROR_TYPE_TEST = 0x1000,
+    OWR_ERROR_TYPE_GST_ERROR,
+    OWR_STATS_TYPE_TEST = 0x2000,
+    OWR_STATS_TYPE_SCHEDULE,
+    OWR_STATS_TYPE_SEND_PIPELINE_ADDED,
+    OWR_STATS_TYPE_SEND_PIPELINE_REMOVED,
+    OWR_EVENT_TYPE_TEST = 0x3000,
+    OWR_EVENT_TYPE_GST_EOS,
```
Comment (superdump) [inline owr/owr_bus.h]: Again, that it is a GStreamer EOS is not interesting to the application, just that some kind of end of stream occurred and a bit more text about what that entails would be good.

File: owr/owr_bus.h [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+/**
+ * OwrMessageSubType:
+ *
+ * @OWR_ERROR_TYPE_GST_ERROR: a GStreamer error message was received
+ * - @debug: #gchar* a string containing debug information
+ * - @code: #gint the error code
+ * - @domain: #gchar the error domain
+ * - @message: #gchar the error message
+ *
+ * @OWR_STATS_TYPE_SCHEDULE: information about a call that was scheduled on the main thread
```
Comment (superdump) [inline owr/owr_bus.h]: Function name?

File: owr/owr_bus.h [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+} OwrMessageType;
+
+/**
+ * OwrMessageSubType:
+ *
+ * @OWR_ERROR_TYPE_GST_ERROR: a GStreamer error message was received
+ * - @debug: #gchar* a string containing debug information
+ * - @code: #gint the error code
+ * - @domain: #gchar the error domain
+ * - @message: #gchar the error message
```
Comment (superdump) [inline owr/owr_bus.h]: Can you translate code and domain to things that do not require gst API knowledge?

File: owr/owr_bus.h [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+ *
+ * @OWR_STATS_TYPE_SCHEDULE: information about a call that was scheduled on the main thread
+ * - @start_time: #gint64 monotonic time when the call was scheduled
+ * - @call_time: #gint64 monotonic time when the function call started
+ * - @end_time: #gint64 monotonic time when the function call completed
+ *
+ * @OWR_STATS_TYPE_SEND_PIPELINE_ADDED: a send pipeline was added to a media session
```
Comment (superdump) [inline owr/owr_bus.h]: Which pipeline?

File: owr/owr_bus.h [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+ * - @start_time: #gint64 monotonic time when the pipeline setup began
+ * - @end_time: #gint64 monotonic time when the pipeline setup was completed
+ *
+ * @OWR_STATS_TYPE_SEND_PIPELINE_REMOVED: a send pipeline was removed to a media session
```
Comment (superdump) [inline owr/owr_bus.h]: Which pipeline?

File: owr/owr_bus.h [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+ * - @start_time: #gint64 monotonic time when the pipeline teardown began
+ * - @end_time: #gint64 monotonic time when the pipeline teardown was completed
+ *
+ * @OWR_EVENT_TYPE_RENDERER_STARTED: a renderer was started
```
Comment (superdump) [inline owr/owr_bus.h]: Which renderer?

File: owr/owr_bus.h [hunk 8]
```diff
@@ -0,0 +1,139 @@
...
+ *
+ * @OWR_EVENT_TYPE_RENDERER_STOPPED: a renderer was stopped
```
Comment (superdump) [inline owr/owr_bus.h]: Which renderer?

**Pull request closed without merging**